src/client/data/config.rs

```diff
@@ -97,9 +98,15 @@ pub struct Config {
     /// The port to connect on.
     #[cfg_attr(feature = "serde", serde(skip_serializing_if = "Option::is_none"))]
     pub port: Option<u16>,
-    /// The password to connect to the server.
+    /// The password to connect to the server. Used with PASS or with SASL authentication
     #[cfg_attr(feature = "serde", serde(skip_serializing_if = "Option::is_none"))]
     pub password: Option<String>,
```

Comment:
You may want to make it a different configuration option. `PASS`'s intended use is a shared server password, that is required to connect to the server at all; while SASL is to authenticate to a specific user account. That `PASS` can also be used as an alternative to SASL to authenticate to user account is an implementation-specific behavior of some servers.

Reply:
I don't see a practical use to having both used at the same time; the "Modern IRC" [document](https://modern.ircdocs.horse/#pass-message) says that SASL is used as alternative to PASS, so I don't think we should see both used in practice. I don't mind adding a second, different option though.

Reply:
I updated the comment to make the intent clear.